### DISCUSSION TITLE
Smart Form Component vs useFormContext

### DISCUSSION THREAD

**Discussion by jgonera** [react-hook-form/react-hook-form#2505] [Q&A]
When would I use a technique demonstrated in https://react-hook-form.com/advanced-usage#SmartFormComponent vs. just using a `FormProvider` and `useFormContext`? Are there any drawbacks to using the context vs. the former technique?

**Comment by bluebill1049** [comment]
why not both? 

Use Smart form when your form is flat and simple, and then use ConnectForm to connect those deeply nested inputs.
